File: 2nd year/1st semester/MAP/ToyLanguageInterpreter/src/Model/Expression/IExpression.java
package Model.Expression;

import Model.Utils.IDictionary;
import Model.Utils.MyDictionary;
import Model.Utils.MyHeap;

public interface IExpression {

    int evaluate(IDictionary<String, Integer> symTable, MyHeap<Integer> heap);

    @Override
    String toString();
}

File: 2nd year/1st semester/MAP/CS_ToyLanguageInterpreter/MAP_Interpreter/Model/Expressions/IExpression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MAP_Interpreter.Model.Expressions
{
    public abstract class IExpression
    {
        public abstract int evaluate(Dictionary<string, int> symTable);

        public abstract override string ToString();
    }
}
